### ISSUE TITLE
RcppParallel problem with reduce_dimension

### ISSUE THREAD

**Issue by scholzmb** [cole-trapnell-lab/monocle3#247]
So, default behavior for reduce_dimension() on my centOS7 server (500GB RAM) segfaults when I run reduce_dimension() without specifying cores, but does not have an issue when core is specified:

> data10X <- reduce_dimension(data10X, core=10)
No preprocess_method specified, and aligned coordinates have been computed previously. Using preprocess_method = 'Aligned'
Note: reduce_dimension will produce slightly different output each time you run it unless you set 'umap.fast_sgd = FALSE' and 'cores = 1'
> data10X <- reduce_dimension(data10X, core=1)
No preprocess_method specified, and aligned coordinates have been computed previously. Using preprocess_method = 'Aligned'

> data10X <- reduce_dimension(data10X)
No preprocess_method specified, and aligned coordinates have been computed previously. Using preprocess_method = 'Aligned'

 *** caught segfault ***
address 0xfffffffffffffff7, cause 'memory not mapped'

Traceback:
 1: RcppParallel::setThreadOptions(numThreads = n_threads)
 2: uwot(X = X, n_neighbors = n_neighbors, n_components = n_components,     metric = metric, n_epochs = n_epochs, alpha = learning_rate,     scale = scale, init = init, init_sdev = init_sdev, spread = spread,     min_dist = min_dist, set_op_mix_ratio = set_op_mix_ratio,     local_connectivity = local_connectivity, bandwidth = bandwidth,     gamma = repulsion_strength, negative_sample_rate = negative_sample_rate,     a = a, b = b, nn_method = nn_method, n_trees = n_trees, search_k = search_k,     method = "umap", approx_pow = approx_pow, n_threads = n_threads,     n_sgd_threads = n_sgd_threads, grain_size = grain_size, y = y,     target_n_neighbors = target_n_neighbors, target_weight = target_weight,     target_metric = target_metric, pca = pca, pca_center = pca_center,     pcg_rand = pcg_rand, fast_sgd = fast_sgd, ret_model = ret_model,     ret_nn = ret_nn, tmpdir = tempdir(), verbose = verbose)
 3: uwot::umap(as.matrix(preprocess_mat), n_components = max_components,     metric = umap.metric, min_dist = umap.min_dist, n_neighbors = umap.n_neighbors,     fast_sgd = umap.fast_sgd, n_threads = cores, verbose = verbose,     nn_method = umap.nn_method, ...)
 4: reduce_dimension(data10X)

obviously, I can work around by always specifying core=1, but it should probably be investigated.

**Comment by scholzmb** [comment]
Hmmm... on further review, memory failure appears to be somewhat random.  Sometimes I get it when specifying core=1, or 10, but not every time.  I cannot reliably troubleshoot yet.

**Comment by panxiaoguang** [comment]
But no matter how many cores I set here, the fault will appear.

**Comment by panxiaoguang** [comment]
> But no matter how many cores I set here, the fault will appear.

Now I know, the problem is caused by RcppParallel package, some compatible trouble brings the bug to me , you can remove this old package and reinstall it .

**Comment by brgew** [comment]
Hi,
Does this problem persist for anyone after reinstalling RcppParallel package?
Thank you for the feedback.

**Comment by hpliner** [comment]
Hopefully fixed by reinstalling RcppParallel - reopen if not
